streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskExecutor.java

```diff
@@ -62,9 +64,16 @@ public TaskExecutor(final Tasks tasks, final ProcessingMode processingMode, fina
      */
     int process(final int maxNumRecords, final Time time) {
         int totalProcessed = 0;
-
-        for (final Task task : tasks.activeTasks()) {
-            totalProcessed += processTask(task, maxNumRecords, time);
+        Task lastProcessed = null;
+        try {
+            for (final Task task : tasks.activeTasks()) {
+                lastProcessed = task;
+                totalProcessed += processTask(task, maxNumRecords, time);
+            }
+        } catch (final Exception e) {
+            tasks.removeTaskFromCuccessfullyProcessedBeforeClosing(lastProcessed);
```

Comment:
If an exception throws here, it means we should have not reached line 93 below and hence this task should not have been added to the set, and hence we do not need to ever call `removeTaskFromCuccessfullyProcessedBeforeClosing` right?